packages/datadog-instrumentations/src/mysql.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const { AsyncResource } = require('async_hooks')
+const {
+  channel,
+  addHook,
+  bind,
+  bindAsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+addHook({ name: 'mysql', file: 'lib/Connection.js', versions: ['>=2'] }, Connection => {
+  const startCh = channel('apm:mysql:query:start')
+  const asyncEndCh = channel('apm:mysql:query:async-end')
+  const endCh = channel('apm:mysql:query:end')
+  const errorCh = channel('apm:mysql:query:error')
+
+  shimmer.wrap(Connection.prototype, 'query', query => function () {
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
+
+    if (!startCh.hasSubscribers || arguments.length === 1 || typeof arguments[arguments.length - 1] !== 'function') {
```

Comment:
I'm not sure to understand this part. If there are no subscribers, why would it still wrap the callback?